Orange/data/util.py

```diff
@@ -95,6 +95,22 @@ def hstack(arrays):
         return np.hstack(arrays)
 
 
+def array_equal(a1, a2):
+    """array_equal that supports sparse and dense arrays with missing values"""
+    if a1.shape != a2.shape:
+        return False
+
+    if not (sp.issparse(a1) or sp.issparse(a2)):  # Both dense: just compare
+        return np.allclose(a1, a2, equal_nan=True)
+
+    v1 = np.vstack(sp.find(a1)).T
+    v2 = np.vstack(sp.find(a2)).T
+    if not (sp.issparse(a1) and sp.issparse(a2)):  # Any dense: order indices
```

Comment:
So, sp.find internally sorts if the matrices have `has_canonical_format=False`, which is False when converting from any sparse matrix. Therefore, sparse matrices currently do not have to be resorted, so your optimization works well. Should we rely on such undocumented behavior, @janezd?